net_test.go

```diff
@@ -0,0 +1,77 @@
+package openssl
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"net"
+	"testing"
+	"time"
+)
+
+func sslConnect(t *testing.T, ssl_listener net.Listener) {
+	for {
+		conn, err := ssl_listener.Accept()
+		if err != nil {
+			t.Errorf("failed accept: %s", err)
+			continue
+		}
+		go func() {
+			io.Copy(ioutil.Discard, io.LimitReader(conn, 1024))
+		}()
+	}
+}
+
+func TestDial(t *testing.T) {
+	ctx := getCtx(t)
+	if err := ctx.SetCipherList("AES128-SHA"); err != nil {
+		t.Fatal(err)
+	}
+	ssl_listener, err := Listen("tcp", "localhost:0", ctx)
+
+	go sslConnect(t, ssl_listener)
```

Comment:
It would be nice to wait for the end of the gorouitine execution (with WaitGroup or a channel).

Reply:
Updated.